### ISSUE TITLE
[css-multicol-2][css-scroll-snap] Snapping to column boxes

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#6017]
It's been brought up a few times before, most recently in https://github.com/w3c/csswg-drafts/issues/5911, that it would be useful to set snapping controls on the individual columns of a multi-column element to enable, for example, using multicol as a pagination mechanism with snapping between "pages".

The most straightforward way to do this would be to add a pseudo-element (e.g. `::column` or `::nth-column(An+B)` that represents each column. It could initially accept only the properties in css-scroll-snap, though we'll likely want to extend it to accept other properties in the future as well.

Thoughts?

p.s. We'll have to contend with also the question of whether to extend this to grid tracks, flex lines, and/or table rows and columns. (It's not so necessary to have this feature there, because they typically have elements we can snap to instead.)

**Comment by rachelandrew** [comment]
Yes, very much a fan of this, the scrolling to a column thing has been mentioned to me a number of times. I like the approach of adding a pseudo-element as that could potentially enable a few other things people want to do.

In terms of extending this to grid tracks and so on, perhaps this is two separate things. One is adding to multicol a pseudo-element so we can target a particular column box, the second is to add the snapping to a identified column/track/flex line.

**Comment by mstensho** [comment]
This is done here: https://drafts.csswg.org/css-multicol-2/#column-pseudo

**Comment by rachelandrew** [comment]
Closing with a link to the resolution that added this https://github.com/w3c/csswg-drafts/issues/10715#issuecomment-2379924982
